### ISSUE TITLE
Cylinder showing problem

### ISSUE THREAD

**Issue by happydpc** [ricosjp/truck#14]
I test the cylinder code following:
```rust
pub fn create_cylinder(height: f64, radius: f64) -> Solid {
    // make a solid cylinder
    let vertex = builder::vertex(Point3::new(0.0, 0.0, radius));
    let circle = builder::rsweep(&vertex, Point3::origin(), Vector3::unit_y(), Rad(7.0));
    let disk = builder::try_attach_plane(&vec![circle]).unwrap();
    let solid = builder::tsweep(&disk, Vector3::new(0.0, height, 0.0));
    // Return the solid as a boundary shell for easier processing later.
    // solid.into_boundaries().pop().unwrap()
    solid
}
```

With default viewer, this comes out this. And some corner shows shadow lines with some camera angle.
![image](https://user-images.githubusercontent.com/771776/129446954-ff382c0c-258b-42c1-a177-e95a7a5ce278.png)

**Comment by ytanimura** [comment]
Thank you very much for your information. I was able to reproduce the problem in my environment. It seems to be a bug when trimming the mesh. We will fix it before the release of v0.3.

**Comment by happydpc** [comment]
I found the problem may be the depth issue. Using the wgpu reverse z-depth method can solve this problem.

**Comment by ytanimura** [comment]
The bug here seems to have already been fixed. I went back through the history and found that the following two commits seemed to have improved the symptoms.

https://github.com/ricosjp/truck/commit/444fdec2b2ff522cd586255b3764ff5dd9a2ce55
https://github.com/ricosjp/truck/commit/65be75cf2532b23f59681483b668d78954d4eea8

As for the bug, it was still a mesh bug. A large triangle overlapping a plane was generated. The same problem occurred with the standard Windows 3D viewer when I read the mesh generated from the commit before the ones above. The reverse depth method probably prevented large triangles from being drawn.

![windows](https://user-images.githubusercontent.com/9394788/144702376-5e370fc0-580d-452f-892d-21f500c7657c.PNG)
